src/validators/union.rs

```diff
@@ -18,12 +18,11 @@ impl UnionValidator {
 
 impl Validator for UnionValidator {
     fn build(schema: &PyDict, config: Option<&PyDict>) -> PyResult<Box<dyn Validator>> {
-        let mut choices: Vec<Box<dyn Validator>> = vec![];
-        let choice_schemas: &PyList = schema.get_as_req("choices")?;
-        for choice in choice_schemas.iter() {
-            choices.push(build_validator(choice, config)?.0);
-        }
-
+        let choices: Vec<Box<dyn Validator>> = schema
+            .get_as_req::<&PyList>("choices")?
+            .iter()
+            .map(|choice| build_validator(choice, config).map(|result| result.0))
+            .collect::<PyResult<Vec<Box<dyn Validator>>>>()?;
```

Comment:
Do you actually need the type here?

Reply:
Yes, otherwise it do not know what to collect too, it maybe could have been reduced to `PyResult<Vec<_>>`, but up to that I would rather have the full signature so it is clear what was intended.

Reply:
no, i think that's fine. Just wondered if it could be inferred, but not in this case.